WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift

```diff
@@ -352,45 +352,46 @@ extension WordPressAuthenticationManager: WordPressAuthenticatorDelegate {
             return
         }
 
-        //Present the epilogue view
+        // Present the epilogue view
         let storyboard = UIStoryboard(name: "LoginEpilogue", bundle: .main)
         guard let epilogueViewController = storyboard.instantiateInitialViewController() as? LoginEpilogueViewController else {
             fatalError()
         }
 
-        epilogueViewController.credentials = credentials
-
-        epilogueViewController.onBlogSelected = { [weak self] blog in
+        let onBlogSelected: ((Blog) -> Void) = { [weak self] blog in
             guard let self = self else {
                 return
             }
 
+            // If the user just signed in, refresh the A/B assignments
+            ABTest.start()
+
             self.recentSiteService.touch(blog: blog)
+            self.presentOnboardingQuestionsPrompt(in: navigationController, onDismiss: onDismiss)
+        }
 
-            guard self.quickStartSettings.isQuickStartAvailable(for: blog) else {
-                if self.windowManager.isShowingFullscreenSignIn {
-                    self.windowManager.dismissFullscreenSignIn(blogToShow: blog)
-                } else {
-                    self.windowManager.showAppUI(for: blog)
-                }
-                return
-            }
+        // If the user has only 1 blog, skip the site selector and go right to the next step
+        guard numberOfBlogs() == 1, let firstBlog = firstBlog() else {
```

Comment:
[nit] I think it would be more readable if we change what we're guarding against. Guard else blocks are rarely that long.

That's what I'm suggesting:
```
// If the user has only 1 blog, skip the site selector and go right to the next step
guard numberOfBlogs() > 1 else {
    if let firstBlog = firstBlog() {
        onBlogSelected(firstBlog)
    }
    return
}

// Continue execution of normal flow (Code that's currently in the else block)
```

Reply:
I think this is an improvement, however with this solution if firstBlog is nil then nothing will happen and the user will be stuck in the login flow. I've updated the code to us an if instead of a guard to achieve the same results as my code, but much cleaner: c59601c51be14c282794efe34b9acf0066e8bd99

Reply:
Yes, that's much better!